src/Infrastructure/Zfs/ZfsFilesystemDTO.php

```diff
@@ -0,0 +1,144 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Infrastructure\Zfs;
+
+final class ZfsFilesystemDTO
+{
+    /**
+     * @var string
+     */
+    private $name;
+
+    /**
+     * @var string
+     */
+    private $available;
+
+    /**
+     * @var string
+     */
+    private $used;
+
+    /**
+     * @var string
+     */
+    private $usedBySnapshot;
+
+    /**
+     * @var string
+     */
+    private $usedByDataset;
+
+    /**
+     * @var string
+     */
+    private $usedByRefreservation;
+
+    /**
+     * @var string
+     */
+    private $usedByChild;
+
+    /**
+     * @var string
+     */
+    private $refer;
+
+    /**
+     * @var string
+     */
+    private $mountPoint;
+
+    /**
+     * @var string
+     */
+    private $origin;
+
+    /**
+     * @var string
+     */
+    private $type;
+
+    public function __construct(
+        string $name,
+        string $available,
+        string $used,
+        string $usedBySnapshot,
+        string $usedByDataset,
+        string $usedByRefreservation,
+        string $usedByChild,
+        string $refer,
+        string $mountPoint,
+        string $origin,
+        string $type
```

Comment:
Should all properties be string ?

Reply:
a src/Infrastructure/Filesystem/BytesFormatConvertor.php class was added to convert strings to byte